instrumentation/host/host.go

```diff
@@ -0,0 +1,279 @@
+type Option interface {
+	// ApplyHost updates *Config.
+	ApplyHost(*Config)
+}
+
+// WithMeterProvider sets the Metric implementation to use for
+// reporting.  If this option is not used, the global metric.Provider
+// will be used.  `provider` must be non-nil.
+func WithMeterProvider(provider metric.Provider) Option {
+	return metricProviderOption{provider}
+}
+
+type metricProviderOption struct{ metric.Provider }
+
+// ApplyHost implements Option.
+func (o metricProviderOption) ApplyHost(c *Config) {
+	c.MeterProvider = o.Provider
+}
+
+var (
+	// Label sets for CPU time measurements.
+
+	LabelCPUTimeUser   = []kv.KeyValue{kv.String("state", "user")}
+	LabelCPUTimeSystem = []kv.KeyValue{kv.String("state", "system")}
+	LabelCPUTimeOther  = []kv.KeyValue{kv.String("state", "other")}
+	LabelCPUTimeIdle   = []kv.KeyValue{kv.String("state", "idle")}
+
+	// Label sets used for Memory measurements.
+
+	LabelMemoryAvailable = []kv.KeyValue{kv.String("state", "available")}
+	LabelMemoryUsed      = []kv.KeyValue{kv.String("state", "used")}
+
+	// Label sets used for Network measurements.
+
+	LabelNetworkTransmit = []kv.KeyValue{kv.String("direction", "transmit")}
+	LabelNetworkReceive  = []kv.KeyValue{kv.String("direction", "receive")}
+)
+
+// Configure computes a Config from the supplied Options.
+func Configure(opts ...Option) Config {
+	c := Config{
+		MeterProvider: global.MeterProvider(),
+	}
+	for _, opt := range opts {
+		opt.ApplyHost(&c)
+	}
+	return c
+}
+
+// Start initializes reporting of host metrics using the supplied Config.
+func Start(c Config) error {
+	if c.MeterProvider == nil {
+		c.MeterProvider = global.MeterProvider()
+	}
+	h := &host{
+		meter: c.MeterProvider.Meter(
+			"host",
+			metric.WithInstrumentationVersion(contrib.SemVersion()),
+		),
+		config: c,
+	}
+	return h.register()
+}
+
+func (h *host) register() error {
+	var (
+		err error
+
+		processCPUTime metric.Float64SumObserver
+		hostCPUTime    metric.Float64SumObserver
+
+		hostMemoryUsage       metric.Int64UpDownSumObserver
+		hostMemoryUtilization metric.Float64UpDownSumObserver
+
+		networkIOUsage metric.Int64SumObserver
+
+		// lock prevents a race between batch observer and instrument registration.
+		lock sync.Mutex
+	)
+
+	proc, err := process.NewProcess(int32(os.Getpid()))
+	if err != nil {
+		return fmt.Errorf("could not find this process: %w", err)
+	}
+
+	lock.Lock()
+	defer lock.Unlock()
+
+	batchObserver := h.meter.NewBatchObserver(func(ctx context.Context, result metric.BatchObserverResult) {
+		lock.Lock()
+		defer lock.Unlock()
+
+		// This follows the OpenTelemetry Collector's "hostmetrics"
+		// receiver/hostmetricsreceiver/internal/scraper/processscraper
+		// measures User and System IOwait time.
+		// TODO: the Collector has per-OS compilation modules to support
+		// specific metrics that are not universal.
+		processTimes, err := proc.TimesWithContext(ctx)
+		if err != nil {
+			global.Handler().Handle(err)
```

Comment:
Same as [this](https://github.com/open-telemetry/opentelemetry-go/blob/fd61d2edece005195116e6b9218801a460a3868d/api/global/handler.go#L90)

```suggestion
			global.Handle(err)
```